core/api/src/app/payments/add-earn.ts

```diff
@@ -29,7 +29,13 @@ export const addEarn = async ({
 }: {
   quizQuestionId: string
   accountId: string
-}): Promise<QuizQuestion | ApplicationError> => {
+}): Promise<
+  | {
+      quizQuestion: Quiz
+      rewardPaymentError?: ApplicationError
```

Comment:
I wonder if it makes sense to wrap this in some like what we've done for the `PartialResult` type and object?

Reply:
I think thats a good idea, I'll wrap it with a Partial result and add a paid field.